src/commands/junit/upload.ts

```diff
@@ -24,6 +26,20 @@ import {buildPath} from '../../helpers/utils'
 const errorCodesNoRetry = [400, 403, 413]
 const errorCodesStopUpload = [400, 403]
 
+const validateXml = (xmlFilePath: string) => {
+  const xmlFileContentString = String(fs.readFileSync(xmlFilePath))
+  const validationOutput = xmlParser.validate(xmlFileContentString)
+  if (validationOutput !== true) {
+    return validationOutput.err.msg
+  }
+  const xmlFileJSON = xmlParser.parse(String(xmlFileContentString))
+  if (!('testsuites' in xmlFileJSON)) {
```

Comment:
Can this be `testsuites` || `testsuite`? some exporters inline a single test suite directly. This would match the parsing we do in backend